src/index.spec.ts

```diff
@@ -1,193 +1,163 @@
-        });
-
-        expect(result.b).toEqual(2);
-        // Prove that mergePartially is a pure function
-        expect(original.b).toEqual(2);
-    })
-
-    it('should overwrite a string even a falsy string', () => {
-        const original = {
-            a: 'a',
-            b: 'b'
-        }
-
-        const result = mergePartially(original, {
-            b: ''
-        });
-
-        expect(result.b).toEqual('');
-        // Prove that mergePartially is a pure function
-        expect(original.b).toEqual('b');
-    })
-
-    it('should allow users to set a value to null if that is something the original type allows', ()=>{
-        const original: IObjWithANullProp = {
-            nonNullProp: 3,
-            nullableProp: 'is not initialized as null'
-        }
-
-        const result = mergePartially(original, {
-            nonNullProp: 4,
-            nullableProp: null
-        })
-
-        expect(result.nonNullProp).toEqual(4),
-        expect(result.nullableProp).toEqual(null);
-        // Prove that mergePartially is a pure function
-        expect(original.nonNullProp).toEqual(3);
-        expect(original.nullableProp).toEqual('is not initialized as null');
-    })
-
-    it('should NOT overwrite a string if it is undefined', () => {
-        const original = {
-            a: 'a',
-            b: 'b'
-        }
-
-        const result = mergePartially(original, {
-            b: undefined
-        });
-
-        expect(result.b).toEqual('b');
-        // Prove that mergePartially is a pure function
-        expect(original.b).toEqual('b');
-    })
-
-    it('is a pure function (i.e. it always returns a copy of the default) even when there is no item passed to merge', ()=>{
-        let original = {
-            a: 'a',
-            b: 'b'
-        }
-
-        const result = mergePartially(original, undefined);
-
-        expect(original).toBe(original);
-        expect(original).not.toBe(result);
+    expect(result.optionalProp).toEqual("a value the original obj didn't have");
+    // Prove that mergePartially is a pure function
+    expect(original.optionalProp).toEqual(undefined);
+  });
+
+  it('supports nested objects', () => {
+    const original = {
+      a: 'a',
+      b: {
+        c: 'c',
+      },
+    };
+
+    const result = mergePartially(original, {
+      b: mergePartially(original.b, {
+        c: 'newC',
+      }),
     });
 
-    it('should add a property that was not initially required', () => {
-        const original: IObjWithOptionalProperty = {
-            requiredProp: "some value"
-        }
-
-        const result = mergePartially(original, {
-            optionalProp: "a value the original obj didn't have"
-        });
-
-        expect(result.optionalProp).toEqual("a value the original obj didn't have");
-        // Prove that mergePartially is a pure function
-        expect(original.optionalProp).toEqual(undefined);
-    })
-
-    it('succeeds where Object.assign does not', () => {
```

Comment:
I was able to get rid of this test due to TypeScript improving Object.assign's type: https://github.com/microsoft/TypeScript/pull/37762